azure-pipelines-v3.yml

```diff
@@ -72,99 +143,74 @@ jobs:
-      dryrun.bot-docs-pr:
-        params: https://github.com/MicrosoftDocs/bot-docs-pr --timeout 40 --no-dry-sync --dry-run --public-template
-      dryrun.azure-docs-pr:
-        params: https://github.com/MicrosoftDocs/azure-docs-pr --profile --timeout 50 --dry-run --regression-rules
+      ${{ each repo in parameters.repos }}:
+        ${{ repo.key }}:
+          params: ${{repo.value.params}}
 
   steps:
-  # Install .NET Core sdk
-  - task: UseDotNet@2
-    displayName: 'Install .NET Core sdk $(dotnetVersion)'
-    inputs:
-      packageType: sdk
-      version: $(dotnetVersion)
-
-  # Download RegressionTest binary
-  - download: current
-    displayName: Download RegressionTest binary
-    artifact: RegressionTest
+    - template: prepare.yml
+      parameters:
+        dotnetVersion: $(dotnetVersion)
 
-  # Install dotnet-trace
-  - script: dotnet tool update -g dotnet-trace
-    displayName: Install dotnet-trace
+    # Run Regression Tests
+    - task: AzureCLI@2
+      displayName: Run
+      inputs:
+        azureSubscription: 'DevRel - Build Agents - 68db5a04-b547-4cfb-a966-6668a3bb3f7c'
+        scriptType: ps
+        scriptLocation: inlineScript
+        powerShellErrorActionPreference: continue
+        inlineScript: $(Pipeline.Workspace)/RegressionTest/docfx.RegressionTest.exe $(params)
+      env:
+        BUILD_REASON: $(Build.Reason)
+        DOCFX_APPDATA_PATH: D:/appdata
+        DOCS_GITHUB_TOKEN: $(gitHubTestToken)
+        DOCS_KV_PROD_ENDPOINT: https://docfxcikvpub.vault.azure.net/
+        AZURE_DEVOPS_TOKEN: $(vsoTestToken)
+        PULL_REQUEST_NUMBER : $(System.PullRequest.PullRequestNumber)
+        APPINSIGHTS_INSTRUMENTATIONKEY: 5226e0d0-086f-4cab-a491-1d00fde9eb09
 
-  # Run Regression Tests
-  - task: AzureCLI@2
-    displayName: Run
-    inputs:
-      azureSubscription: 'DevRel - Build Agents - 68db5a04-b547-4cfb-a966-6668a3bb3f7c'
-      scriptType: ps
-      scriptLocation: inlineScript
-      powerShellErrorActionPreference: continue
-      inlineScript: $(Pipeline.Workspace)/RegressionTest/docfx.RegressionTest.exe $(params)
-    env:
-      BUILD_REASON: $(Build.Reason)
-      DOCFX_APPDATA_PATH: D:/appdata
-      DOCS_GITHUB_TOKEN: $(gitHubTestToken)
-      DOCS_KV_PROD_ENDPOINT: https://docfxcikvpub.vault.azure.net/
-      AZURE_DEVOPS_TOKEN: $(vsoTestToken)
-      PULL_REQUEST_NUMBER : $(System.PullRequest.PullRequestNumber)
-      APPINSIGHTS_INSTRUMENTATIONKEY: 5226e0d0-086f-4cab-a491-1d00fde9eb09
+- job: WarmUp
+  timeoutInMinutes: 180
+  condition: and(succeeded(), eq(variables['pipeline-option'], 'warm-up'))
+  dependsOn: WindowsBuild
+  pool:
+    name: Docfx-1ES
+  strategy:
+    matrix:
+      ${{ each agent in parameters.agents }}:
+        ${{ agent.key }}:
+          agentIndex: ${{ agent.value }}
+  steps:
+    - template: prepare.yml
+      parameters:
+        dotnetVersion: $(dotnetVersion)
+    # Warm up
+    - task: PowerShell@2
+      displayName: create needed directory
```

Comment:
Can this be done in `regressiontest.exe`?

Reply:
Yes, but I think do it outside of the regressontest.exe should be clear. Or we will try to create this dir multiple times.